packages/layout/src/slots/Target.tsx

```diff
@@ -1,59 +1,83 @@
-import { useClassName, useId } from '@contember/react-utils'
-import { assert, dataAttribute, isNonEmptyTrimmedString } from '@contember/utilities'
+import { useClassName, useId, useReferentiallyStableValue } from '@contember/react-utils'
+import { assert, dataAttribute, isArrayOfMembersSatisfyingFactory, isNonEmptyArray, isNonEmptyTrimmedString, isSingleWordString, satisfiesOneOfFactory, setHasOneOf } from '@contember/utilities'
+import { snakeCase } from 'change-case'
 import { memo, useLayoutEffect, useRef } from 'react'
-import slugify from 'slugify'
 import { useActiveSlotPortalsContext, useTargetsRegistryContext } from './contexts'
-import { TargetProps } from './types'
+import { SlotTargetProps } from './types'
 
 export type OwnTargetContainerProps = {
-	'data-has-own-children': boolean;
-	'data-name': string;
 	className: string;
 }
 
+const isNameString = satisfiesOneOfFactory(
+	isNonEmptyTrimmedString,
+	isSingleWordString,
+)
+
+const nonEmptyArrayOfNonEmptyStrings = satisfiesOneOfFactory(
+	isNonEmptyArray,
+	isArrayOfMembersSatisfyingFactory(isNameString),
+)
+
 /**
  * @group Layout
  */
-export const Target = memo<TargetProps>(
+export const Target = memo<SlotTargetProps>(
 	({
 		as,
+		aliases,
 		componentClassName = 'slot',
 		className: classNameProp,
-		children,
+		fallback,
 		name,
 		...rest
 	}) => {
-		assert('name is non-empty string', name, isNonEmptyTrimmedString)
+		assert('name is non-empty string without spaces', name, isNameString)
+
+		aliases = aliases ?? [name]
+		assert('aliases is non-empty string or array of non-empty strings', aliases, nonEmptyArrayOfNonEmptyStrings)
+
+		const names = useReferentiallyStableValue(aliases)
```

Comment:
why not just sth like
```
const names = useRef(aliases)
names.current = aliases
```

Reply:
There is an `useLayoutEffect()` depending on `names` just few lines later below, it needs to be a stable value reference otherwise it would cause unnecessary re-registrations of the same slot targets on each re-render.

Reply:
i know, my solution is still ref stable. it is just more simple

Reply:
I understand, I considered the same solution too. Does it cover all legitimate effect runs during the lifespan of the component?

Reply:
On a side note:

When using the `aliases` prop, the required `name` is left mostly for consistent targeting of the node in CSS and does not register as target for `name` prop. To do that, it needs to be also placed as member of aliases. After some thinking, this might be contra-intuitive and I think it also needs to register target for the `name` prop as well.

Reply:
i've added comment for that function, it seems totally wrong to me.

btw, if you want to have stable array by value, you can use sth like
```
const aliasesStable = useMemo(() => aliases, [...aliases])
```

Reply:
I rewrote that part. I am keeping a list and I register/unregister what is needed, finally I unregister rest on component unmount. Works with all scenarios.